From f2582ae8422037d18cafc7891daad6a4d744aa39 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Mon, 22 Aug 2022 18:15:31 +0200
Subject: [PATCH 071/954] fix(stm32mp1): check DDR refresh to restore firewall
 configuration

When returning from STANDBY mode, we must not re-apply the firewall
configuration as some DDR part has been reconfigured by the BL32
with some other configuration linked to specific features.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2feb816dafd836a71a0fa4fa1e18207780ce763b fix(stm32mp1): check DDR refresh to restore firewall configuration]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I4ed42618f583c3243c5ca9e0c9b1b29cd4fcfd6a
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/263522
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp1/stm32mp1_fconf_firewall.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/plat/st/stm32mp1/stm32mp1_fconf_firewall.c b/plat/st/stm32mp1/stm32mp1_fconf_firewall.c
index 7d9956409a..d981ec4844 100644
--- a/plat/st/stm32mp1/stm32mp1_fconf_firewall.c
+++ b/plat/st/stm32mp1/stm32mp1_fconf_firewall.c
@@ -10,6 +10,7 @@
 #include <common/fdt_wrappers.h>
 #include <drivers/arm/tzc400.h>
 #include <drivers/clk.h>
+#include <drivers/st/stm32mp1_ram.h>
 #include <dt-bindings/clock/stm32mp1-clks.h>
 #include <lib/fconf/fconf.h>
 #include <lib/object_pool.h>
@@ -56,6 +57,11 @@ void stm32mp1_security_setup(void)
 {
 	uint8_t i;
 
+	/* DDR content will be restored, do not change the firewall protection */
+	if (stm32mp1_ddr_is_restored()) {
+		return;
+	}
+
 	assert(nb_regions > 0U);
 
 	tzc400_init(STM32MP1_TZC_BASE);
@@ -85,6 +91,11 @@ static int fconf_populate_stm32mp1_firewall(uintptr_t config)
 	/* Assert the node offset point to "st,mem-firewall" compatible property */
 	const char *compatible_str = "st,mem-firewall";
 
+	/* DDR content will be restored, do not change the firewall protection */
+	if (stm32mp1_ddr_is_restored()) {
+		return 0;
+	}
+
 	node = fdt_node_offset_by_compatible(dtb, -1, compatible_str);
 	if (node < 0) {
 		ERROR("FCONF: Can't find %s compatible in dtb\n", compatible_str);
-- 
2.53.0

